core/src/main/java/com/graphhopper/util/CHEdgeIteratorState.java

```diff
@@ -43,6 +43,16 @@
      */
     boolean isShortcut();
 
+    /**
+     * @return true if this shortcut can be used in fwd direction. Do not call this method if {@link #isShortcut()} is false
+     */
+    boolean getFwdAccess();
+
+    /**
+     * @see #getFwdAccess
+     */
+    boolean getBwdAccess();
```

Comment:
These methods are needed / used to get the access flags for shortcuts which are different to the access flags of normal edges, because they cannot be 'changed' by a configurable weighting (after the preparation has finished)

Reply:
These methods are only necessary for node-based CH at the moment, because for edge-based CH the edges are assumed to be always fwd==true and bwd==false, is my understanding correct here?

Reply:
Yes this is true and also pretty interesting. There are lots of shortcuts with fwd=bwd=true and this way we can store two weights with one double. The shortcuts either have the same weight for both directions or they are unidirectional, in both cases there is only one weight. For edge-based CH we do not do this optimization yet. Possibly it also will not work as well as it does for node-based because of turn costs at the skipped nodes (they usually depend on the travel direction), but of course doing this could be an improvement (50% less memory in the best (albeit very unlikely) case).